### PR TITLE
JAMES-2633 FlagsUpdated event constructor simplification

### PR DESCRIPTION
**Pull request by trantienduchn**
FlagsUpdated constructor only need list of `UpdatedFlags`, the list of `MessageUid` can be computed by updatedflags list

### CODE REVIEW

File: mailbox/api/src/main/java/org/apache/james/mailbox/MailboxListener.java [hunk 1]
```diff
@@ -411,10 +412,12 @@ public final int hashCode() {
         private final List<MessageUid> uids;
         private final List<UpdatedFlags> updatedFlags;
 
-        public FlagsUpdated(MailboxSession.SessionId sessionId, User user, MailboxPath path, MailboxId mailboxId, List<MessageUid> uids, List<UpdatedFlags> updatedFlags) {
+        public FlagsUpdated(MailboxSession.SessionId sessionId, User user, MailboxPath path, MailboxId mailboxId, List<UpdatedFlags> updatedFlags) {
```
Comment (chibenwa) [inline mailbox/api/src/main/java/org/apache/james/mailbox/MailboxListener.java]: You don't need modifications in EventFactory?

File: mailbox/store/src/main/java/org/apache/james/mailbox/store/event/EventFactory.java [hunk 2]
```diff
@@ -59,12 +59,8 @@
...
     }
 
-    public MailboxListener.FlagsUpdated flagsUpdated(MailboxSession session, List<MessageUid> uids, Mailbox mailbox, List<UpdatedFlags> uflags) {
-        return flagsUpdated(session.getSessionId(), session.getUser(), uids, mailbox, uflags);
-    }
-
-    public MailboxListener.FlagsUpdated flagsUpdated(MailboxSession.SessionId sessionId, User user, List<MessageUid> uids, Mailbox mailbox, List<UpdatedFlags> uflags) {
-        return new MailboxListener.FlagsUpdated(sessionId, user, mailbox.generateAssociatedPath(), mailbox.getMailboxId(), uids, uflags);
+    public MailboxListener.FlagsUpdated flagsUpdated(MailboxSession session, Mailbox mailbox, List<UpdatedFlags> uflags) {
+        return new MailboxListener.FlagsUpdated(session.getSessionId(), session.getUser(), mailbox.generateAssociatedPath(), mailbox.getMailboxId(), uflags);
```
Comment (trantienduchn) [inline mailbox/store/src/main/java/org/apache/james/mailbox/store/event/EventFactory.java]: @chibenwa it is here

File: mailbox/api/src/main/java/org/apache/james/mailbox/MailboxListener.java (hunk 1 above)
Comment (trantienduchn) [reply mailbox/api/src/main/java/org/apache/james/mailbox/MailboxListener.java]: it is here https://github.com/linagora/james-project/pull/2055/commits/a071ef5cae186f69ca982af3905e13199d60d378#r242465375

File: mailbox/api/src/main/java/org/apache/james/mailbox/MailboxListener.java [hunk 3]
```diff
@@ -37,6 +37,7 @@
 import org.apache.james.mailbox.model.QuotaRoot;
```
Comment (chibenwa) [inline mailbox/api/src/main/java/org/apache/james/mailbox/MailboxListener.java]: No JIRA ticket prefix in commit message?

**Comment by chibenwa** [comment]
test this please

**Comment by aduprat** [comment]
test this please

**Comment by chibenwa** [comment]
test this please

**Comment by trantienduchn** [comment]
```
[a071ef5cae186f69ca982af3905e13199d60d378] [ERROR] Errors: 
[a071ef5cae186f69ca982af3905e13199d60d378] [ERROR]   VacationIntegrationTest.jmapVacationShouldNotSendNotificationTwice:240 » ConditionTimeout
[a071ef5cae186f69ca982af3905e13199d60d378] [INFO] 
[a071ef5cae186f69ca982af3905e13199d60d378] [ERROR] Tests run: 655, Failures: 0, Errors: 1, Skipped: 2
```

**Comment by trantienduchn** [comment]
```
[1999e529761db35d9876c7d47aff8431e33b25c0] [ERROR] Errors: 
[1999e529761db35d9876c7d47aff8431e33b25c0] [ERROR]   SwiftKeystone3ObjectStorageBlobsDAOBuilderTest.builtBlobsDAOCanStoreAndRetrieve » Completion
[1999e529761db35d9876c7d47aff8431e33b25c0] [INFO] 
[1999e529761db35d9876c7d47aff8431e33b25c0] [ERROR] Tests run: 124, Failures: 0, Errors: 1, Skipped: 0
```
test this please

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**